Libraries/LLM/Phi.swift

```diff
@@ -73,14 +73,12 @@ private class PhiAttention: Module {
             keys = rope(keys)
         }
 
-        queries = queries.asType(Float.self)
-        keys = keys.asType(Float.self)
-
         // Finally perform the attention computation
         let scale = sqrt(1 / Float(queries.dim(-1)))
         let output = MLXFast.scaledDotProductAttention(
-            queries: queries, keys: keys, values: values, scale: scale, mask: mask
+            queries: queries.asType(.float32), keys: keys, values: values, scale: scale, mask: mask
```

Comment:
I found that if I passed `queries` in as Float16 I would get unimaginative output

<img width="720" alt="image" src="https://github.com/ml-explore/mlx-swift-examples/assets/46639364/2377aae2-aab2-48fa-9e26-7c0d2bc95d19">

Is this a reasonable thing to do here?  Should keys be cast to float32 as well (it didn't seem to matter when I tested it)?

@awni

Reply:
So if you upcast the queries the full attention should be done in fp32 and it should work. The keys will be implicitly cast in the op.

The way you’ve done it looks good to me. Most models don’t need this, phi2 is an exception for some reason